### ISSUE TITLE
Empty LLP shown when not logged in

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#5768]
### Description

![image](https://user-images.githubusercontent.com/1190097/33519281-47bfa1d0-d761-11e7-8bc4-e27c96a45c17.png)

### Version information

- **Platform**: web (in-browser)
- **Browser**: Chrome 62 (incognito)
- **OS**: Windows 10
- **URL**: riot.im/develop

**Comment by lampholder** [comment]
We should think about how we want to guide users into this feature, bearing in mind that an empty lump of screen looks weird.

**Comment by turt2live** [comment]
Considering the feature is behind a labs flag and being shown to new users is a bit concerning.

**Comment by turt2live** [comment]
@lampholder Can this be squeezed into 0.14? Either in the capacity of the LLP coming out of labs or ensuring the LLP isn't shown when the LLP isn't enabled.

**Comment by lampholder** [comment]
We've decided for now that LLP will be 'on by default', but can be disabled with a user setting (but only after you've logged in).

This probably isn't 100% optimal in terms of user journey (since it could take a little while to discover what that left, empty side bar is all about), but it is 'good enough' for now and we can iterate on it :)

**Comment by turt2live** [comment]
This bug is specifically that it is shown when **not** logged in.

**Comment by lampholder** [comment]
Yeah, understood :) My point was that because it defaults to 'on', showing when not logged in is actually 'correct' - you're seeing the view you'd see (or as close as makes sense) immediately after making a new account (just like it defaults to the light theme, etc.).

**Comment by turt2live** [comment]
/me is confused, but trusts that someone knows about this and it'll get fixed in a sane way

**Comment by lampholder** [comment]
I wonder if the missing detail is that this isn't in labs any more - it has now graduated to being a proper feature, configurable via a user setting (which defaults to 'on').

So yes, it would have (probably) have been a bug to display a labs feature to not-logged-in users, but since it's not in labs any more I don't think that it's a bug.

My reasoning is that I think we're presenting the app to not-logged-in users as best we can. Bearing in mind we know nothing about them, I think it makes sense to show not-logged-in users the app with controls in the **default** or **initial** configuration. This means: default (light green) theme, empty FAVOURITES, PEOPLE, ROOMS, LOW PRIORITY and HISTORICAL room headers, and yes, an empty LLP.

Does that address your confusion or am I still wide of the mark? Thanks for your patience btw :)

**Comment by turt2live** [comment]
It is true that this was opened before the tag panel went always-on, however the [setting still exists](https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/settings/Settings.js#L97-L102) which allows custom riot deployments to put it into labs/disabled, therefore reproducing this bug (still). It seems as though the riot.im deployment instead has it toggled to `enabled` which gets rid of the labs setting in the UI, but does not prevent other deployments from changing the behaviour.

TLDR: It's still in labs by nature of the setting still being present.

**Comment by lampholder** [comment]
Okay, I think maybe I finally understand. 😅 

The LLP's being displayed is conditional today upon two things:
- the LLP feature being enabled in config.json
- the 'Hide community filter panel' setting (in Settings.js) being set to true

And the bug is that, for any given instance of riot, the visibility of LLP for a not-logged-in user should depend on the values/default values of those two things, rather than just always being visible. Is that correct?

Additionally, I think we could help ourselves by agreeing some shared terminology around the labs stuff. I'd like to propose:

- _behind a feature flag_ == feature visibility is dependent upon a value in config.js
- _in labs_ == feature visibility is _additionally_ dependent upon a user setting in the 'Labs' section of the UX

(suggestions/feedback welcome)

**Comment by turt2live** [comment]
> And the bug is that, for any given instance of riot, the visibility of LLP for a not-logged-in user should depend on the values/default values of those two things, rather than just always being visible. Is that correct?

Yes. If the feature flag were removed and all we were left with was the user setting, then this can be closed.

**Comment by turt2live** [comment]
Closing: LLP's feature flag was removed
